### ISSUE TITLE
Overriden method stub in a generic subclass uses wrong generic parameter type

### ISSUE THREAD

**Issue by SolalPirelli** [dotnet/roslyn#13425]
**Version Used**: VS 2015 Update 3 (14.0.25425.01)

**Steps to Reproduce**:

1 - Write the following code, where `|` is the caret:
```
class B<X>
{
    public virtual void M( X x ) { }
}

class D<Y> : B<Y>
{
     override|
}
```

2 - Type a space, to trigger suggestions for overriding methods.

3 - Select the `M(Y x)` item.

**Expected Behavior**: The method stub should have the signature `M(Y x)`.

**Actual Behavior**: The method stub has the signature `M(X x)`, which does not compile.

**Comment by Pilchie** [comment]
@CyrusNajmabadi - does this look familiar to you?  Did we have a similar bug around implement abstract class?

**Comment by CyrusNajmabadi** [comment]
That doesn't look at all familiar (and is super weird).  I'm happy to take a look though!

**Comment by jmarolf** [comment]
fixed in 2.0.0.6090704
![intellisense_fix](https://cloud.githubusercontent.com/assets/9797472/18304104/5a57c262-7495-11e6-99f1-c76737f5a28b.gif)
